forge/forge/op/eval/common.py

```diff
@@ -164,6 +164,12 @@ def eval_debug_print(type, inputs, output):
 
 
 def calculate_pcc(a, b):
+    # broadcasting single value tensor to 2 value tensor to support pcc calculation
```

Comment:
Why do we force PCC on single value tensors? For scalar values, we can't calculate PCC, but we can compare golden and calculated values using relative and absolute tolerances:
```
 torch.allclose(golden, calculated)
```

Reply:
Note that calculate_pcc is called from inside compare_with_golden_pcc which should only be used for non scalar values. compare_with_golden function handles both scalar and non scalar values...

Reply:
@dgolubovicTT  If that is a case, can we replace all occurrence of `compare_with_golden_pcc` with `compare_with_golden` as it handles more cases like boolean, scalars along with `compare_with_golden_pcc` ?

What do you think @nvukobratTT ?

Reply:
I added a new commit for this in this PR. Can you please review it now? @dgolubovicTT

Reply:
yeah sure @dgolubovicTT . Let's here it from @vkovinicTT

Reply:
Yeah we should ideally use only `compare_with_golden` as it handles all the cases. But my current [pr](https://github.com/tenstorrent/tt-forge-fe/pull/739) is about unifying all of the verification, where you only call `verify(input, framework_model, compile_model) `and it will perform `compare_with_golden` alongside other checks so you don't have to worry about data format or anything. I hope that pr will be merged by tomorrow.